crates/router/src/connector/bluesnap/transformers.rs

```diff
@@ -0,0 +1,304 @@
+use serde::{Deserialize, Serialize};
+use crate::{
+    core::errors,
+    pii::{self, PeekInterface, Secret},
+    types::{self, api, storage::enums},
+};
+
+//TODO: Fill the struct with respective fields
+#[derive(Default, Debug, Serialize, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct BluesnapPaymentsRequest {
+    amount: String,
+    credit_card: Card,
+    currency: String,
+    soft_descriptor: Option<String>,
+    card_transaction_type: BluesnapPaymentStatus,
```

Comment:
Try to differentiate BluesnapPaymentStatus and BluesnapType
And to handle response based on type refer to [authorizedotnet](https://github.com/juspay/hyperswitch/tree/authorize-net-features/crates/router/src/connector/authorizedotnet)

Here I have created a tuple and sent it while handling the response in `try_from()` you will be able to find it and implement the same here